File: app/src/main/java/com/pkfcooperparry/template/ui/form/FormActivityModule.java
package com.pkfcooperparry.template.ui.form;

import com.pkfcooperparry.template.data.DataManager;
import com.pkfcooperparry.template.utils.rx.SchedulerProvider;

import dagger.Module;
import dagger.Provides;

@Module
public class FormActivityModule {

    @Provides
    FormViewModel provideVisitViewModel(DataManager dataManager, SchedulerProvider schedulerProvider) {
        return new FormViewModel(dataManager, schedulerProvider);
    }
}

File: app/src/main/java/com/pkfcooperparry/template/ui/form/FormActivityModule.kt
package com.pkfcooperparry.template.ui.form

import com.pkfcooperparry.template.data.DataManager
import com.pkfcooperparry.template.utils.rx.SchedulerProvider

import dagger.Module
import dagger.Provides

@Module
class FormActivityModule {

    @Provides
    internal fun provideVisitViewModel(dataManager: DataManager, schedulerProvider: SchedulerProvider): FormViewModel {
        return FormViewModel(dataManager, schedulerProvider)
    }
}
